Sources/Recycling/ListViews/UserAds/Cell/UserAdsListViewActiveCell.swift

```diff
@@ -0,0 +1,202 @@
+            detailLabel.topAnchor.constraint(equalTo: priceLabel.bottomAnchor, constant: .mediumSpacing),
+        ])
+    }
+
+    private func setupRibbonView(forUserAdStatus status: UserAdStatus) {
+        ribbonView?.removeFromSuperview()
+        ribbonView = nil
+
+        switch status {
+        case .started: ribbonView = RibbonView(style: .warning, with: status.rawValue)
+        case .active: ribbonView = RibbonView(style: .success, with: status.rawValue)
+        case .inactive: ribbonView = RibbonView(style: .disabled, with: status.rawValue)
+        case .expired: ribbonView = RibbonView(style: .disabled, with: status.rawValue)
+        case .sold: ribbonView = RibbonView(style: .warning, with: status.rawValue)
+        case .unknown: return
+        }
+
+        if let view = ribbonView {
+            view.translatesAutoresizingMaskIntoConstraints = false
+            addSubview(view)
+            NSLayoutConstraint.activate([
+                view.centerYAnchor.constraint(equalTo: accessoryView?.centerYAnchor ?? contentView.centerYAnchor),
+                view.trailingAnchor.constraint(equalTo: accessoryView?.leadingAnchor ?? contentView.trailingAnchor),
+            ])
+        }
+    }
+
+    // MARK: - Superclass Overrides
+
+    public override func prepareForReuse() {
+        super.prepareForReuse()
+        adImageView.image = nil
+        titleLabel.text = nil
+        priceLabel.text = nil
+        detailLabel.text = nil
+        ribbonView = nil
+        accessibilityLabel = nil
+
+        if let model = model {
+            dataSource?.userAdsListViewActiveCell(self, cancelLoadingImageForModel: model, imageWidth: adImageView.frame.size.width)
+        }
+    }
+
+    // MARK: - Dependency injection
+
+    public var model: UserAdsListViewModel? {
+        didSet {
+            guard let model = model else { return }
+            titleLabel.text = model.title
+            priceLabel.text = model.price
+            detailLabel.text = model.detail
+            setupRibbonView(forUserAdStatus: UserAdStatus(rawValue: model.status) ?? .unknown)
+            accessibilityLabel = model.accessibilityLabel
+        }
+    }
+
+    // MARK: - Public
+
+    func loadImage() {
+        if let model = model {
+            loadImage(model)
+        }
+    }
+
+    /// Loads a given image provided that the imagePath in the `model` is valid.
+
+    private func loadImage(_ model: UserAdsListViewModel) {
+        guard let dataSource = dataSource, model.imagePath != nil else {
+            loadingColor = .clear
+            adImageView.image = defaultImage
+            return
+        }
+
+        adImageView.backgroundColor = loadingColor
+
+        dataSource.userAdsListViewActiveCell(self, loadImageForModel: model, imageWidth: frame.size.width) { [weak self] image in
+            self?.adImageView.backgroundColor = .clear
+
+            if let image = image {
+                self?.adImageView.image = image
```

Comment:
How do you feel about
`self?.adImageView.image = image ?? self?.defaultImage`
instead?

Reply:
Good idea👍